python/ray/scripts/scripts.py

```diff
@@ -1848,6 +1854,260 @@ def local_dump(
+            [
+                print(f"{api_endpoint}/v1/api/logs/file/{node_id}/{log}")
+                for log in links[key]
+            ]
+
+    for lang in ray_constants.LANGUAGE_WORKER_TYPES:
+        print_section(f"{lang.capitalize()} Core Driver Logs", f"{lang}_driver_logs")
+        print_section(
+            f"{lang.capitalize()} Core Worker Logs", f"{lang}_core_worker_logs"
+        )
+    print_section("Worker Errors", "worker_errors")
+    print_section("Worker Stdout", "worker_outs")
+    print_section("Raylet Logs", "raylet_logs")
+    print_section("GCS Logs", "gcs_logs")
+    print_section("Miscellaneous Logs", "misc")
+    print_section("Autoscaler Monitor Logs", "autoscaler_monitor")
+    print_section("Dashboard Logs", "dashboard")
+    print_section("Ray Client Logs", "ray_client")
+
+
+@cli.command()
+@click.argument(
+    "filters",
+    nargs=-1,
+    required=False,
+    default=None,
+)
+@click.option(
+    "--filename",
+    "-f",
+    type=str,
+    required=False,
+    default=None,
+    help="Specify the exact filename of the log file.",
+)
+@click.option(
+    "-ip",
+    "--node-ip",
+    required=False,
+    type=str,
+    default=None,
+    help="Filters the logs by this ip address.",
+)
+@click.option(
+    "--node-id",
+    "-n",
+    required=False,
+    type=str,
+    default=None,
+    help="Filters the logs by this NodeID.",
+)
+@click.option(
+    "--pid",
+    "-pid",
+    required=False,
+    type=str,
+    default=None,
+    help="Retrieves the logs from the process with this pid.",
+)
+@click.option(
+    "--actor-id",
+    "-a",
+    required=False,
+    type=str,
+    default=None,
+    help="Retrieves the logs corresponding to this ActorID.",
+)
+@click.option(
+    "--task-id",
+    "-t",
+    required=False,
+    type=str,
+    default=None,
+    help="Retrieves the logs corresponding to this TaskID.",
+)
+@click.option(
+    "--watch",
+    "-w",
+    required=False,
+    type=bool,
+    is_flag=True,
+    help="Streams the log file as it is updated instead of just tailing.",
+)
+@click.option(
+    "--lines",
+    "-l",
+    required=False,
+    type=int,
+    default=None,
+    help="Number of lines to tail from log. -1 indicates fetching the whole file.",
+)
+def logs(
+    filters,
+    filename: str,
+    node_ip: str,
+    node_id: str,
+    pid: str,
+    actor_id: str,
+    task_id: str,
+    watch: bool,
+    lines: int,
+):
+    """
+    View logs in the ray cluster.
+
+    FILTERS: Filename substrings (filename, component, file extension, id) to filter
+    the logs by.
+
+    Example usage:
+
+    ray logs --actor-id=XYZ              # Display stdout log by actor-id
+
+    ray logs --task-id=XYZ               # Display stdout log by task-id
+
+    ray logs --ip=198.0.0.1 --pid=987    # Display worker stdout by ip & pid
+
+    ray logs -n <node-id> -f raylet.out  # Display log by filename & node-id
+
+    ray logs                             # Display list of logs by category
```

Comment:
Two API qusetions.

1. What's happening if we just call ray logs? Does it return the head node list or all log files? (Personally prefer the former).
2. If there's only one match, does it print the log or the index?

Reply:
I feel that its too verbose and confusing to document the behaviour in the CLI help. Users should discover this behaviour, it feels intuitive IMO.